aeternity/helpers.go

```diff
@@ -47,111 +65,67 @@ func GetNextNonce(c getAccounter, accountID string) (nextNonce uint64, err error
-	PostTransaction(string, string) (err error)
+// GetAccountsByName returns any account_pubkey entries that it finds in a name's Pointers.
+func (h Helpers) GetAccountsByName(name string) (addresses []string, err error) {
+	return h.getAnythingByName(name, "account_pubkey")
 }
 
-// BroadcastTransaction differs from Client.PostTransaction() in that the latter just handles
-// the HTTP request via swagger, the former recalculates the txhash and compares it to the node's
-//  response after POSTing the transaction.
-func BroadcastTransaction(c transactionPoster, txSignedBase64 string) (err error) {
-	// Get back to RLP to calculate txhash
-	txRLP, _ := Decode(txSignedBase64)
-
-	// calculate the hash of the decoded txRLP
-	rlpTxHashRaw, _ := hash(txRLP)
-	// base58/64 encode the hash with the th_ prefix
-	signedEncodedTxHash := Encode(PrefixTransactionHash, rlpTxHashRaw)
+// GetOraclesByName returns any oracle_pubkey entries that it finds in a name's Pointers.
+func (h Helpers) GetOraclesByName(name string) (oracleIDs []string, err error) {
+	return h.getAnythingByName(name, "oracle_pubkey")
+}
 
-	// send it to the network
-	err = c.PostTransaction(txSignedBase64, signedEncodedTxHash)
-	return
+// GetContractsByName returns any contract_pubkey entries that it finds in a name's Pointers.
+func (h Helpers) GetContractsByName(name string) (contracts []string, err error) {
+	return h.getAnythingByName(name, "contract_pubkey")
 }
 
-// PrintGenerationByHeight utility function to print a generation by it's height
-// TODO this belongs in cmd and needs to be tested with error cases
-func (c *Node) PrintGenerationByHeight(height uint64) {
-	r, err := c.GetGenerationByHeight(height)
-	if err == nil {
-		PrintObject("generation", r)
-		// search for transaction in the microblocks
-		for _, mbh := range r.MicroBlocks {
-			// get the microblok
-			mbhs := fmt.Sprint(mbh)
-			r, err := c.GetMicroBlockTransactionsByHash(mbhs)
-			if err != nil {
-				Pp("Error:", err)
-			}
-			// go through all the hashes
-			for _, btx := range r.Transactions {
-				p, err := c.GetTransactionByHash(fmt.Sprint(btx.Hash))
-				if err == nil {
-					PrintObject("transaction", p)
-				}
-			}
-		}
-	} else {
-		fmt.Println("Something went wrong in PrintGenerationByHeight")
-	}
+// GetChannelsByName returns any channel entries that it finds in a name's Pointers.
+func (h Helpers) GetChannelsByName(name string) (channels []string, err error) {
+	return h.getAnythingByName(name, "channel")
 }
 
 // Context stores relevant context (node connection, account address) that one might not want to spell out each time one creates a transaction
 type Context struct {
-	Client  *Node
 	Address string
+	Helpers HelpersInterface
 }
 
 // NewContext ensures that every field of a Context is filled out.
-func NewContext(client *Node, address string) Context {
-	return Context{Client: client, Address: address}
+func NewContext(address string, helpers HelpersInterface) Context {
```

Comment:
create helpers within this one, so one does not have to create a helpers struct and pass it into the context.